packages/api/tests/fieldCrop.test.js

```diff
@@ -0,0 +1,562 @@
+const chai_should = chai.should();  // Using Should style
+const server = require('./../src/server');
+const dummy = require('./dummy');
+const sinon = require('sinon')
+const Knex = require('knex')
+const environment = 'test';
+const config = require('../knexfile')[environment];
+const knex = Knex(config);
+let checkJwt;
+jest.mock('jsdom')
+jest.mock('../src/middleware/acl/checkJwt')
+const mocks  = require('./mock.factories');
+
+describe('FieldCrop Tests', () => {
+  let middleware;
+  let newOwner;
+  let newManager;
+  let newWorker;
+  let newUnregisteredWorker;
+  let newUser;
+  let crop;
+  let field;
+  let farm;
+  let farmNewUser;
+  let fieldCrop;
+  let cropNotInUse;
+  beforeAll(() => {
+    token = global.token;
+  });
+
+  function postFieldCropRequest( data, callback, user_id = newOwner.user_id, farm_id = farm.farm_id) {
+    chai.request(server).post('/field_crop')
+      .set('Content-Type', 'application/json')
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .send(data)
+      .end(callback)
+  }
+
+  function postCropRequest( data, callback, user_id = newOwner.user_id, farm_id = farm.farm_id) {
+    chai.request(server).post('/crop')
+      .set('Content-Type', 'application/json')
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .send(data)
+      .end(callback)
+  }
+
+  function getRequest(url, callback,user_id = newOwner.user_id, farm_id = farm.farm_id) {
+    chai.request(server).get(url)
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .end(callback)
+  }
+
+  function putFieldCropRequest(data, callback, user_id = newOwner.user_id, farm_id = farm.farm_id, field_crop_id = fieldCrop.field_crop_id) {
+    chai.request(server).put(`/field_crop/${field_crop_id}`)
+      .set('farm_id', farm_id)
+      .set('user_id', user_id)
+      .send(data)
+      .end(callback)
+  }
+
+  function deleteRequest(url, callback,user_id = newOwner.user_id, farm_id = farm.farm_id) {
+    chai.request(server).delete(url)
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .end(callback)
+  }
+
+  function fakeUserFarm(role=1){
+    return ({...mocks.fakeUserFarm(),role_id:role});
+  }
+  function fakeFieldCrop(){
+    const fieldCrop = mocks.fakeFieldCrop();
+    const area_used = fieldCrop.area_used < field.area?  fieldCrop.area_used: field.area;
+    return ({...fieldCrop, crop_id: crop.crop_id, field_id: field.field_id, area_used});
+  }
+  function fakeCrop(farm_id = farm.farm_id){
+    const crop = mocks.fakeCrop();
+    return ({...crop, farm_id});
+  }
+
+  const later = async (delay) =>
+    new Promise(resolve => setTimeout(resolve, delay));
+
+  beforeEach(async () => {
+    await knex.raw(`
+    DELETE FROM "fieldCrop";
+    DELETE FROM "field";
+    DELETE FROM "userFarm";
+    DELETE FROM "crop";
+    DELETE FROM "farm";
+    DELETE FROM "users";
+    DELETE FROM "weather_station";
+    `);
```

Comment:
Is this still necessary?

Reply:
afterAll is only triggered after all tests finish. Cleanup before each tests is needed so that tests are "unit tests"

Reply:
Oh I didnt notice it was afterAll, I thought it was afterEach, that was the way we had it before right?

Reply:
Yes. The behavior of afterEach is weird. Sometimes it is triggered at the end of post/delete request. Sometimes it is not triggered.

Reply:
@JimmyRowland lets make a Tech debt story about this!